test/model/src/include/model/core.h

```diff
@@ -0,0 +1,437 @@
+ */
+class data_value {
+
+public:
+    /*
+     * data_value::data_value --
+     *     Create a new instance.
+     */
+    inline data_value(const char *data) : _data(data) {}
+
+    /*
+     * data_value::data_value --
+     *     Create a new instance.
+     */
+    inline data_value(const std::string &data) noexcept : _data(data) {}
+
+    /*
+     * data_value::data_value --
+     *     Create a new instance.
+     */
+    inline data_value(const std::string &&data) noexcept : _data(std::move(data)) {}
+
+    /*
+     * data_value::create_none --
+     *     Create an instance of a "None" value.
+     */
+    inline static data_value
+    create_none() noexcept
+    {
+        return data_value();
+    }
+
+    /*
+     * data_value::as_string --
+     *     Return the data value as a human-readable string (e.g., for printing).
+     */
+    inline const std::string &
+    as_string() const noexcept
+    {
+        return _data.has_value() ? *_data : NONE_STRING;
+    }
+
+    /*
+     * data_value::operator== --
+     *     Compare to another data value.
+     */
+    inline bool
+    operator==(const data_value &other) const noexcept
+    {
+        return _data == other._data;
+    }
+
+    /*
+     * data_value::operator!= --
+     *     Compare to another data value.
+     */
+    inline bool
+    operator!=(const data_value &other) const noexcept
+    {
+        return !(*this == other);
+    }
+
+    /*
+     * data_value::operator< --
+     *     Compare to another data value.
+     */
+    inline bool
+    operator<(const data_value &other) const noexcept
+    {
+        return _data < other._data;
+    }
+
+    /*
+     * data_value::operator<= --
+     *     Compare to another data value.
+     */
+    inline bool
+    operator<=(const data_value &other) const noexcept
+    {
+        return _data <= other._data;
+    }
+
+    /*
+     * data_value::operator> --
+     *     Compare to another data value.
+     */
+    inline bool
+    operator>(const data_value &other) const noexcept
+    {
+        return !(*this <= other);
+    }
+
+    /*
+     * data_value::operator> --
+     *     Compare to another data value.
+     */
+    inline bool
+    operator>=(const data_value &other) const noexcept
+    {
+        return !(*this < other);
+    }
+
+    /*
+     * data_value::tombstone --
+     *     Check if this is a None value.
+     */
+    inline bool
+    none() const noexcept
+    {
+        return !_data.has_value();
+    }
+
+private:
+    /*
+     * data_value::data_value --
+     *     Create a new instance.
+     */
+    inline data_value() : _data(std::nullopt) {}
+
+    std::optional<std::string> _data;
+};
+
+/*
+ * NONE --
+ *     The "None" value.
+ */
+extern const data_value NONE;
+
+/*
+ * kv_update --
+ *     The data value stored in a KV table, together with the relevant update information, such as
+ *     the timestamp.
+ */
+class kv_update {
```

Comment:
I think the components of the system would be easier to understand if each class lived in its own `.h`.

Reply:
Done. I have a habit of combining, but I can see how what I had could have been confusing.